### ISSUE TITLE
Ignore `<template>` in `vue/html-self-closing` rule

### ISSUE THREAD

**Issue by Thomasan1999** [vuejs/eslint-plugin-vue#2202]
**What rule do you want to change?**
`vue/html-self-closing`

**Does this change cause the rule to produce more or fewer warnings?**
N/A

**How will the change be implemented? (New option, new default behavior, etc.)?**
Either by default or add an option to ignore `<template>` the same way `<style>` and `<script>` tags are ignored. Empty `<template>` tags don't make sense and should either be fully-removed from the Vue template by the engineer or left alone (if that's the way the engineer prefers it).

**Please provide some example code that this change will affect:**

```vue
<CustomComponent>
    <template #slotName></template>
</CustomComponent>
```

```vue
<template></template>
```

**What does the rule currently do for this code?**

```vue
<CustomComponent>
    <template #slotName />
</CustomComponent>
```

```vue
<template />
```

**What will the rule do after it's changed?**
It would leave the code unaffected.

**Additional context**
N/A

**Comment by FloEdelmann** [comment]
Hmm, I'm not sure why `<template></template>` should be a special case. What is the purpose?
Did you check whether the [current options](https://eslint.vuejs.org/rules/html-self-closing.html#options) of the [`vue/html-self-closing`](https://eslint.vuejs.org/rules/html-self-closing.html) rule are maybe already sufficient?

Note that there is the (recommended) [`vue/no-lone-template`](https://eslint.vuejs.org/rules/no-lone-template.html#vue-no-lone-template) rule that disallows empty `template` elements anyway.

**Comment by Dimava** [comment]
This is a problem when you use `autofix on save`
I often make templates like
```vue
<script setup lang="ts">
</script>
<template>
</template>
```
and having it to self-close the `<template>` on save is annoying
I have to either fix that manually (which is annoying) or not to save (which is even more annoying)

As there is [`vue/no-lone-template`](https://eslint.vuejs.org/rules/no-lone-template.html#vue-no-lone-template) empty `<template></template>` is considered invalid, so `<template/>` is obviously invalid, so autofixing unfinished `<template></template>` to invalid `<template/>` makes no sense

**Comment by FloEdelmann** [comment]
I think that top-level block tags should just all be excluded in `vue/html-self-closing`. A pull request is welcome for this change.

However, I don't think that other `<template>` elements should then get any special treatment.

**Comment by Dimava** [comment]
Or `excludeRoot: boolean` as I don't think collapsing root items may ever make sense

**Comment by FloEdelmann** [comment]
As I said [previously](https://github.com/vuejs/eslint-plugin-vue/issues/2202#issuecomment-1595838849), top-level block tags should just all be excluded (without options). A pull request is welcome for this change.
